d3-hexbin/d3-hexbin-test.ts

```diff
@@ -0,0 +1,211 @@
+/**
```

Comment:
As a general note, since the tests in DT are strictly "shape tests", there are a few patterns I tend to follow in the other D3 module definitions:
* When defining variables/constants always explicitly type them with the _expected_ type. I.e. `let x : number = ...` rather than `let x = ...` This way, tests will fail if the right-hand  side does not meet the expectation. Otherwise, the definitions could appear internally consistent due to TypeScripts implicit type inference. The same holds for accessor functions to be tested as the return type of a getter.
* Test each signature for each interface method
* For a "setter" method signature always test the return type individually. Do not use consecutive chaining in shape tests. Otherwise, the test may not find edge cases where a return type is (accidentally) changed by one method, but the next chained method is also defined on it, e.g. a sub-type. Re-assigning to the object on which a setter was invoked is a quick check for chainability.
* As in the case of `hexbin()`, there are sometimes generator factories, which make a default assumption about underlying data types (like each point  is of type `[number, number]`), but allow for more general data structures. In this case by setting the generic `T` e.g. to type `Point`. Again, I would first define a variable with explicit type, like `Hexbin<[number, number]>` or `Hexbin<Point`>, then test the factory by assigning to those variables.
* When there are accessors involved, like here for `x` and `y`, I tend to test against each of the typed generators I obtained from the factory.

One aspect, that is helpful is creating explicit tests for expected failure modes, e.g. assigning `let x: d3Hexbin.Hexbin<Point> = d3Hexbin.hexbin();`. Same for e.g. method signatures with accessor functions depending on a generic. The benefit is that it may catch some edge cases. Note that such tests can only be inspected by you when updating tests. As you should see compiler errors in your IDE right away. You will need to comment them out before committing, so the CI tests do not fail. I tend to do that by also appending a comment like `// fails, argument type mismatch` or something similar.

You can find examples of the latter e.g. in d3-selection.